### DISCUSSION TITLE
Map en vuex

### DISCUSSION THREAD

**Discussion by GloriaCoralCerecedo** [LaunchX-InnovaccionVirtual/FrontEnd-Mision#554] [Q&A]
Buenas...sabes cómo se usa el .map en vuex con los arreglos (?)

**Comment by HugoHerreraR** [comment]
depende de como esta estructurado tu map, pero normalmente un arreglo es declarado como  "arreglo: [ ]" y para llenarlo se utiliza el push

**Reply by GloriaCoralCerecedo** [reply]
Oh ya veo eso va en el script?

**Reply by codingcodax** [reply]
Si, pero el `script` no tiene que tener `setup`.
Quedaría algo así:
```vue
<script>
export default {
  data() {
    return {
      orders: [{}, {}, ...]
    }
  }
}
</script>
```

**Reply by GloriaCoralCerecedo** [reply]
Oh bueno lo checaré

**Comment by codingcodax** [comment]
Si estas usando `<script>` puedes hacerlo así:
```vue
<script>
export default {
  data() {
    return {
      orders: [{}, {}, ...]
    }
  }
}
</script>
```
Pero si estas usando `<script setup>` no tienes que hacer el `export default`.
```vue
<script setup>
const orders = [{}, {}, ...]
</script>
```

**Reply by GloriaCoralCerecedo** [reply]
Gracias lo checaré, todo eso se hace en el componente no?

**Reply by codingcodax** [reply]
Si.

Por ejemplo:

```vue
// FormSection.vue

<script setup>
...
</script>

<template>
...
</template>

<style>
...
</style>
```

**Reply by GloriaCoralCerecedo** [reply]
Ok gracias

**Reply by GloriaCoralCerecedo** [reply]
Es que todo está en el state, entonces en el componente adornos pues hay que llamarlo

**Reply by codingcodax** [reply]
```js
// store.js
import { reactive } from 'vue'

export const store = reactive({
  products: [{}, {}, ...]
})
```

```vue
<!-- ComponentA.vue -->
<script>
import { store } from './store.js'

export default {
  data() {
    return {
      products
    }
  }
}
</script>
```

**Reply by GloriaCoralCerecedo** [reply]
Muchas gracias

**Reply by GloriaCoralCerecedo** [reply]
Me pasas las paginas donde sacas esto o es la pagina oficial de vuew

**Reply by codingcodax** [reply]
[State Management](https://vuejs.org/guide/scaling-up/state-management.html#state-management)
